tests/test_initialization.py

```diff
@@ -300,11 +309,86 @@ def test_lora_pissa_conversion_same_output_after_loading(self, data, tmp_path):
         )
 
         # save the model with conversion
+        peft_model.save_pretrained(
+            tmp_path / "pissa-model-converted", path_initial_model_for_weight_conversion=tmp_path / "init-model"
+        )
+        model_converted = PeftModel.from_pretrained(deepcopy(model), tmp_path / "pissa-model-converted")
+        output_converted = model_converted(data)[0]
+
+        assert torch.allclose(output_pissa, output_converted, atol=tol, rtol=tol)
+        # rank should be double of what it was initially
+        assert model_converted.peft_config["default"].r == 16
+        assert model_converted.base_model.model.linear.lora_A["default"].weight.shape[0] == 16
+        # base model weights should be the same as the initial model
+        assert torch.allclose(
+            model.linear.weight, model_converted.base_model.model.linear.base_layer.weight, atol=tol, rtol=tol
+        )
+
+    def test_lora_pissa_conversion_same_output_after_loading_deprecated(self, data, tmp_path):
+        model = self.get_model()
+        config = LoraConfig(init_lora_weights="pissa", target_modules=["linear"], r=8)
+        peft_model = get_peft_model(deepcopy(model), config)
+        peft_model.peft_config["default"].init_lora_weights = True
+        peft_model.save_pretrained(tmp_path / "init-model")
+        peft_model.peft_config["default"].init_lora_weights = "pissa"
+
+        tol = 1e-06
+        peft_model.base_model.linear.lora_B["default"].weight.data *= 2.0
+        output_pissa = peft_model(data)[0]
+
+        # test the warning
```

Comment:
Hmm, I don't quite get this test. From the name and the comment, it looks like it's supposed to check the deprecation warning, but I don't see any such check. Instead, the test seems to be basically a copy of the previous test.

Reply:
Maybe my comment is misleading. I am actually testing that when we specify the deprecated option is the behaviour same with the new option (i.e did we correctly re-assign and update the option).

Reply:
Okay I see. In this case, let's remove this comment. Let's rename the test function to `test_lora_pissa_conversion_same_output_after_loading_with_deprecated_arg` and also add a `# TODO: remove test for deprecated arg in PEFT v0.14.0` comment.

Furthermore, let's add a test below to check the warning itself (which was what I initially thought this test was doing). You can use `with pytest.warns` or the `recwarn` fixture for this. Also add the `TODO` comment there.